src/components/filters/filter-dynamic/filter-dynamic.component.ts

```diff
@@ -85,13 +101,15 @@ export class FilterDynamicComponent implements OnInit, OnDestroy {
                     this.selected = filter;
                 }
             });
+
+            this._changeDetector.markForCheck();
         });
 
         // get the items to be displayed in the typeahead
         this.typeaheadItems = this.getItems();
 
         // determine if we should show the typeahead control
-        if (this.options && this.options.maxIndividualItems && this.options.maxIndividualItems + 1 >= this.filters.length) {
+        if (this.options?.maxIndividualItems && this.options.maxIndividualItems + 1 >= this.filters.length) {
```

Comment:
I personally think if statements should be easy to skim read and therefore be able to understand the purpose of the check without needing to delve into the actual logic. In cases like this, I think it would be beneficial to extract this into a well named variable that explains its purpose (which also removes the need for the comment above). Maybe something like `isTypeaheadRequired`? Then it would be `if (!isTypeaheadRequired)` here which I think makes it much easier to quickly understand the purpose and intent of the logic.

Reply:
I have shortened the condition as part of it was unnecessary. I have moved it to a variable called `isTypeaheadHidden`. We are setting a `showTypeahead` variable so as this is the inverse the opposite would be `hide` or `hidden`. 

While I agree, if the code is clear then we should remove the comment, in this case I have updated the comment, as even with the variable name the logic can be hard to understand without context.

Reply:
I'm not sure if `isTypeaheadHidden` is accurate in this situation. `isTypeaheadHidden` implies to me that we're checking what the current state of the typeahead is but here we're actually checking if we need to hide the typeahead. So to me this currently reads as "if the typeahead is hidden then hide the typeahead" (which makes no sense) instead of "if the typeahead should be hidden, then hide the typeahead". Maybe something like `shouldHideTypeahead` would be better?

I agree that with the current variable name it's difficult to understand the context without the comment. But I think in general we should be aiming to remove the need for comments by using descriptive variable names. Maybe something like `numberOfFiltersExceedsMaxIndividualItems` (which while it's a long name, it does convey the purpose behind the if statement, which would then read as "if the number of filters exceeds the max individual items, then hide the typeahead"). I'm not sure, I'm fine if you want to stick with the updated comment for now though.

Reply:
I'm happy to change it to `shouldHideTypeahead`. I'm not against long variable names where they bring value, however in this case the problem is more that `maxIndividualItems` probably wasn't a good name in the first place but we can't change that now. The term 'Individual items' does not make it obvious that these are the items that will always be visible and not in the typeahead list so I think its easier for someone who may not know the component very well to have the brief explanation here.

Reply:
Sounds good, I agree `maxIndividualItems` is probably the issue here. Think the current version is definitely clearer than what it initially was, so I'm happy with that 👍